### PR TITLE
Added message when leaving filter subdialog with the condition not added

### LINKED ISSUE #5208: Information message when closing filter sub dialog without adding a condition
A common issue for users is when they are defining a filter they select the column and set the condition values but don't always know to click the "Add Condition" button.

We could have a really easy help for this by having an information pop up when you click Return and there is a column selected in the receiver. When you click "Add Condition" it clears to receiver so likely if the receiver if filled the user may have forgot to click the button.

I did something like this recently on the Import NetCDF file subdialog - this could have the same information style message box as there.

### PR DESCRIPTION
**Pull request by Muthenya**
Fixes #5208.
@africanmathsinitiative/developers this is ready for review.

### CODE REVIEW

File: instat/sdgCreateFilter.vb [hunk 1]
```diff
@@ -36,4 +36,10 @@ Public Class sdgCreateFilter
     Private Sub sdgCreateFilter_Load(sender As Object, e As EventArgs) Handles Me.Load
         autoTranslate(Me)
     End Sub
+
+    Private Sub sdgCreateFilter_Closing(sender As Object, e As CancelEventArgs) Handles Me.Closing
+        If Not ucrCreateFilter.ucrFilterByReceiver.IsEmpty Then
+            MessageBox.Show(text:="Warning. See details below." & Environment.NewLine & "Are you sure you want to return to the main dialog?" & Environment.NewLine & "Condition is not yet added", caption:="Missing information", buttons:=MessageBoxButtons.YesNo, icon:=MessageBoxIcon.Information)
```
Comment (Muthenya) [inline instat/sdgCreateFilter.vb]: I wasn't sure on the message. Any ideas @africanmathsinitiative/developers?
@dannyparsons I set the condition focusing on ucrFilterByReceiver. Wouldn't be more efficient if  we focused on the Add Condition button being enabled or disabled? Just a thought.

Comment (dannyparsons) [reply instat/sdgCreateFilter.vb]: How about:

> Are you sure you want to return to the main dialog?
> The condition for [column name] has not been added. Click the "Add Condition" button if you want to add it.

Comment (Muthenya) [reply instat/sdgCreateFilter.vb]: This is done.

**Comment by Muthenya** [comment]
This is now ready.

File: instat/sdgCreateFilter.vb [hunk 2]
```diff
@@ -36,4 +37,15 @@ Public Class sdgCreateFilter
     Private Sub sdgCreateFilter_Load(sender As Object, e As EventArgs) Handles Me.Load
         autoTranslate(Me)
     End Sub
+
+    Private Sub sdgCreateFilter_Closing(sender As Object, e As CancelEventArgs) Handles Me.Closing
+        Dim result As MsgBoxResult
+
+        If Not ucrCreateFilter.ucrFilterByReceiver.IsEmpty Then
+            result = MessageBox.Show("Are you sure you want to return to the main dialog?" & Environment.NewLine & "The condition for " & ucrCreateFilter.ucrFilterByReceiver.GetVariableNames(False) & " has not been added." & Environment.NewLine & "Click the " & Chr(34) & "Add Condition" & Chr(34) & " button if you want to add it.", caption:="Missing information", buttons:=MessageBoxButtons.YesNo, icon:=MessageBoxIcon.Information)
```
Comment (dannyparsons) [inline instat/sdgCreateFilter.vb]:
```suggestion
            result = MessageBox.Show("Are you sure you want to return to the main dialog?" & Environment.NewLine & "The condition for " & ucrCreateFilter.ucrFilterByReceiver.GetVariableNames(False) & " has not been added." & Environment.NewLine & "Click the " & Chr(34) & "Add Condition" & Chr(34) & " button if you want to add it.", caption:="Return to main dialog", buttons:=MessageBoxButtons.YesNo, icon:=MessageBoxIcon.Information)
```

Just a suggestion on the dialog caption.

Comment (Muthenya) [reply instat/sdgCreateFilter.vb]: Though I feel it repeats the first line of, it is much better, perhaps with a "?".

Comment (dannyparsons) [reply instat/sdgCreateFilter.vb]: Yes that sounds fine. I'm not really sure what sensible titles are for these so open to suggestions.

Comment (Muthenya) [reply instat/sdgCreateFilter.vb]: I have made the change.

**Pull request merged**